### PR TITLE
Fix frontend unit tests

### PR DESCRIPTION
**Pull request by brojd**
# What this PR does

- bring back and fix frontend unit tests

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] Added the relevant release notes label (see labels prefixed w/ `release:`). These labels dictate how your PR will
    show up in the autogenerated release notes.

### CODE REVIEW

File: grafana-plugin/src/containers/PluginConfigPage/PluginConfigPage.test.tsx [hunk 1]
```diff
@@ -197,39 +196,6 @@ describe('PluginConfigPage', () => {
     expect(component.container).toMatchSnapshot();
   });
 
-  test("If onCallApiUrl is not set in the plugin's meta jsonData, and ONCALL_API_URL is passed in process.env, it calls selfHostedInstallPlugin", async () => {
```
Comment (brojd) [inline grafana-plugin/src/containers/PluginConfigPage/PluginConfigPage.test.tsx:200]: It's tricky to fix after we changed `getOnCallApiUrl` fn to check if it's running in the browser, we would need to mimic that `window` is `undefined` etc, I decided to remove those 2 unit tests for now

File: grafana-plugin/src/models/user/user.test.ts [hunk 2]
```diff
@@ -1,54 +0,0 @@
-import { makeRequest as makeRequestOriginal } from 'network/network';
```
Comment (brojd) [inline grafana-plugin/src/models/user/user.test.ts:1]: this test suite doesn't make much sense after we switched to typed http client

**Review by teodosii — APPROVED** [review]

**Review by matiasb — APPROVED** [review]

File: .github/workflows/linting-and-tests.yml [hunk 3]
```diff
@@ -69,9 +69,9 @@ jobs:
         if: steps.cache-frontend-dependencies.outputs.cache-hit != 'true'
         working-directory: grafana-plugin
         run: yarn install --frozen-lockfile --prefer-offline --network-timeout 500000
-      - name: Build frontend (will run linter and tests)
+      - name: Build, lint and test frontend
         working-directory: grafana-plugin
-        run: yarn build
+        run: yarn lint && yarn test && yarn build
```
Comment (joeyorlando) [inline .github/workflows/linting-and-tests.yml:74]: previously `yarn build` used to run the linting and tests, I guess w/ the move to the new plugin toolkit build tool version that was no longer the case?

**Review by joeyorlando — COMMENTED** [review]
niceee 🙌

**Pull request merged**